libs/gl-client/src/lnurl/pay/mod.rs

```diff
@@ -123,6 +173,47 @@ fn ensure_amount_is_within_range(
     Ok(())
 }
 
+//LUD-16: Paying to static internet identifiers.
+pub fn parse_lightning_address(lightning_address: &str) -> Result<String> {
+    let lightning_address_components: Vec<&str> = lightning_address.split("@").collect();
+
+    let username = lightning_address_components.get(0);
+
+    if username.is_none() {
+        return Err(anyhow!("Could not parse username in lightning address"));
+    }
+
+    let path = username.unwrap();
+
+    let domain = lightning_address_components.get(1);
+
+    if domain.is_none() {
+        return Err(anyhow!("Could not parse domain in lightning address"));
+    }
+
+    let domain = domain.unwrap();
+
+    if !validate_domain(domain) {
+        return Err(anyhow!("Domain {} is not a valid", domain));
+    }
+
+    let pay_request_url = ["https://", domain, "/.well-known/lnurlp/", path].concat();
+    return Ok(pay_request_url);
+}
+
+fn validate_domain(domain: &str) -> bool {
+    match Regex::new(
+        r"^(?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?\.)+[a-z0-9][a-z0-9-]{0,61}[a-z0-9][.]?$",
+    ) {
+        Ok(regex) => {
+            return regex.is_match(domain);
+        }
+        Err(_e) => {
+            return false;
+        }
+    }
+}
```

Comment:
I'm not sure we need an additional validation here: since we're using `reqwest` to parse the URL later, and it may still be happy with the `host` part of the URL, it should also be able to perform a DNS lookup for that host. Adding additional validation here seems more risky than warranted (reqwest will tell us if it was unable to resolve and contact a host anyway), and may introduce false negatives (UTF-8 domains are not accepted here, as are emojis, and I'm not quite convinced that IPv4 and IPv6 addresses would be accepted either). Please remove this, and the dependency on regex.

Reply:
This is intentional. UTF-8 characters in domains should *not* be accepted as there are ascii conversion standards for that. For example, the domain for '😂.com' would resolve to 'xn--g28h.com' which would pass validation here. Ip addresses are also not domains and aren't valid for lightning addresses so wouldn't be considered here either. 

That being said, I'll remove this function and leave validation/resolution for the reqwest library when actually making the request.

Reply:
Removed